src/Plugin/jsonapi/FieldEnhancer/CardLinkEnhancer.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+namespace Drupal\tide_landing_page\Plugin\jsonapi\FieldEnhancer;
+
+use Drupal\tide_media\Plugin\jsonapi\FieldEnhancer\ImageEnhancer;
+use Drupal\jsonapi_extras\Plugin\ResourceFieldEnhancerBase;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\taxonomy\Entity\Term;
+use Drupal\media\Entity\Media;
+use Drupal\file\Entity\File;
+use Shaper\Util\Context;
+
+/**
+ * Adds necessary fields from internal link.
+ *
+ * @ResourceFieldEnhancer(
+ *   id = "card_link_enhancer",
+ *   label = @Translation("Card Link Enhancer(Adds necessary fields from internal link)"),
+ *   description = @Translation("Adds necessary fields from internal link.")
+ * )
+ */
+class CardLinkEnhancer extends ResourceFieldEnhancerBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function doUndoTransform($data, Context $context) {
+    if (!empty($data) && (strpos($data['uri'], 'entity:node/') !== FALSE)) {
+      $nid = str_replace("entity:node/", "", $data['uri']);
+      if (!empty($nid)) {
+        $data['url'] = $this->getAlias($nid);
+        $data['image'] = $this->getImage($nid);
+        $data['internal_node_fields'] = $this->getCardFields($nid);
+      }
+    }
+    return $data;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function doTransform($data, Context $context) {
+    return $data;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getOutputJsonSchema() {
+    return [
+      'anyOf' => [
+        ['type' => 'array'],
```

Comment:
The output can be any of below?

Reply:
ah it should be array, object, and string

Reply:
@asarfara updated to object.